### ISSUE TITLE
Smack user in a face with release notes after update

### ISSUE THREAD

**Issue by zawor** [microsoft/terminal#5513]
# Description of the new feature/enhancement

Smack user in a face with release notes dialog/view/whatever you feel like appropriate after each update. It should be rather opt in feature as I can imagine not everybody wishes to see those. 

It should solve (to some extent) situations like #5458 where breaking changes had limited visibility until user got hit with fallout. And those which run on Windows Store version (like I do) and don't hold their hand on the pulse of latest upcoming changes (just like me) were pretty surprised by it.

# Proposed technical implementation details (optional)

In perfect world I would love to see it 1-to-1 like VS Code does. Love how they did release notes and call me pervert but I read them from cover to cover every single time. You simply get smacked with new tab with clear and well organized release notes in new tab every time you get updated. I imagine that built in dedicated view like in VS Code may be not quite feasible in reasonable time/budget but opening browser window pointing to release notes will do the job too.

**Comment by DHowett-MSFT** [comment]
Yeah, we definitely need a better on-upgrade story. I think it'd be really cool if we popped it up in an _actual terminal tab_. Like, using the terminal renderer and stuff.

**Comment by zadjii-msft** [comment]
_@Rosefield:_
> It would be cool if when the terminal upgrades it showed you the changes in the new version(s). This would help with feature discovery since many users would not know/think to look at the blog posts or github release notes.
> 
> With #11308 there will be some precedence for showing non-terminal content in the terminal, and that infrastructure could potentially be used to show a modal/textbox with the release notes.
> 
> With the ApplicationState it can also store the last used version, so that a diff between old ~ new could be showed.
